packages/block-library/src/cover/style.scss

```diff
@@ -6,7 +6,7 @@
 	background-position: center center;
 	min-height: 430px;
 	width: 100%;
-	margin: 0 0 1.5em 0;
+	margin: 0 0 2em 0;
```

Comment:
When this margin is 2em, it perfectly matches the existing margin. This means no overlap between blocks when selected or hovered:

![Screenshot 2019-03-13 at 13 09 14](https://user-images.githubusercontent.com/1204802/54277866-40b9be80-4591-11e9-8e53-b746ef8ea6ba.png)

We can't totally prevent that from happening. For example a 3rd party block might add zero margin, and it'll look like this:

![Screenshot 2019-03-13 at 13 10 06](https://user-images.githubusercontent.com/1204802/54277913-56c77f00-4591-11e9-96d7-f6d528fff54b.png)

I.e. overlap. 

Is this something we should be okay with? Technically it works fine since the multi block selection style and block boundaries are painted outside the block. But surfacing here for discussion.

Reply:
This style rule will only match up with the current margins for the bottom.
Is matching up with the current margins a goal with all blocks, some blocks , or just by coincidence sometimes?

Reply:
> This style rule will only match up with the current margins for the bottom.
Is matching up with the current margins a goal with all blocks, some blocks , or just by coincidence sometimes?

Well it's a little more complex than that. In https://codepen.io/joen/pen/XOLamR you can see sort of a minimalistic microcosm of the current block setup. The red backgrounds represent the _block footprint_. The margin between the red is what those blocks themselves provide. You could have zero margin, and the text would bump against each other. Since blocks paint the selected borders _outside_ the block footprint, this would mean those borders would overlap adjacent blocks, like this:

![Screenshot 2019-03-15 at 10 51 56](https://user-images.githubusercontent.com/1204802/54423227-7933d680-4710-11e9-948a-6b2fd814626b.png)

Inversely if you have a ton of margin, you'll just get a lot of space between blocks:

![Screenshot 2019-03-15 at 10 52 06](https://user-images.githubusercontent.com/1204802/54423300-a3859400-4710-11e9-891b-4f2c75dec277.png)

There's $block-padding variable, set to 14px if I recall. It would be nice to, separately, explore making this variable styleable by an editor style. For example if you have a layout that relies on really tight block margins but you don't want the selected block borders to overlap. But this is a challenge, and one best looked at separately I'd say.

So to answer your question, yes it's a lucky coincidence that 2em = 32px (2x16px base font), and that happens to be the same as `($block-padding * 2) + $block-spacing;`

Reply:
Is the border really a border? Perhaps it should use outline instead, so it doesn't affect the margin collapse.